### PR TITLE
renamed files and added dockstring

### PR DESCRIPTION
**Pull request by anpolol**
train_model не трогала, его возможно переписывать надо, а те ошибки линтера которые есть я не знаю как исправить. Некоторые типы не знала как называть, комментарием писала где ээто было

### CODE REVIEW

File: stable_gnn/embedding/model_train_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            self.loss,
+        )
+        # np.save('../data_help/embedings_'+str(self.dataset_name)+str(self.loss['name'])+'.npy', out.cpu().numpy())
+
+        # scheduler.step()
+
+        return out
+
+
+class _OptunaTrainEmbeddings(_ModelTrainEmbeddings):
```
[the remark this answers is not in the document]
Comment (danwhale) [reply stable_gnn/embedding/model_train_embeddings.py]: _SamplerAPP еще =)

[the remark this answers is not in the document]
Comment (danwhale) [reply stable_gnn/embedding/model_train_embeddings.py]: Хотя, слушай, я сделаю абстрактный сэмплер в след коммите и сам допишу докстринги)

File: stable_gnn/embedding/model_train_embeddings.py [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-
 import optuna
 import torch
 import torch_geometric.transforms as T
 from torch_geometric.loader import NeighborSampler
 
 from stable_gnn.graph import Graph
+from torch import device
+from stable_gnn.embedding.model import Net
+from .sampling import Sampler
```
Comment (danwhale) [inline stable_gnn/embedding/model_train_embeddings.py:9]: Относительные импорты - это очень плохо)

File: tests/test_general.py [hunk 3]
```diff
@@ -50,21 +50,21 @@
     )
 
     # best_values = {'hidden_layer': 32, 'dropout': 0.0, 'size of network, number of convs': 3, 'lr': 0.001,"number of negative samples for graph.adjust":5}
-    model, test_acc_mi, test_acc_ma = M.run(best_values)
+    model, test_acc_mi, test_acc_ma = model_training.run(best_values)
```
Comment (danwhale) [inline tests/test_general.py:53]: Мы можем какой-то ассерт здесь впилить? Нужен хотябы один рабочий смоук тест. 

На accuracy можно сделать что-то такое: 

```python 
assert np.isclose(test_acc, expected_accuracy, tol=0.05)
```

File: tests/test_general.py [hunk 4]
```diff
@@ -50,21 +50,21 @@
...
-        A = np.load(root + name + "/A.npy")
+        adj_matrix = np.load(root + name + "/A.npy")
     except:
         A = torch.squeeze(to_dense_adj(data.edge_index)).numpy()
 
-    explainer = Explain(model=model, A=A, X=X)
+    explainer = Explain(model=model, adj_matrix=adj_matrix, features=features)
 
     pgm_explanation = explainer.structure_learning(28)
     print("explanations is 1", pgm_explanation.nodes(), pgm_explanation.edges())
```
Comment (danwhale) [inline tests/test_general.py:70]: На вывод обьяснений тоже

File: stable_gnn/model_gc.py [hunk 5]
```diff
@@ -9,20 +6,42 @@
...
+    def __init__(
+        self,
+        dataset: List[Graph],
+        device: device,  # Я не знаю как правильно тут  тип обозначить и выше тоже
+        hidden_layer: int = 64,
+        dropout: int = 0,
+        num_layers: int = 2,
+        ssl_flag: bool = False,
+        heads: int = 1,
+        **kwargs
```
[the remark this answers is not in the document]
Comment (danwhale) [reply stable_gnn/model_gc.py:43]: это еще актуально

File: stable_gnn/model_gc.py [hunk 6]
```diff
@@ -54,52 +73,65 @@ def __init__(self, dataset, device, conv="GAT", hidden_layer=64, dropout=0, num_
...
+        """
+        Adjust dataset so that to increase extrapolation ability
+
+        :param train_indices: ([int]): List of train indices
+        :param val_indices: ([int]): List of validation indices
+        :param init_edges: (bool): If True, there would be a list of init edges as start for Learning structure of Bayesian Net(default:'False')
+        :param remove_init_edges: If True, it is possible that edges from init_list would be removed during the structure learning of Bayesian Net(default:'False')
+        :param white_list: If True, edges inBayesian Net would be only from this white list (default:'False')
+        :param score_func: (str): Name of score function to optimize, either 'MI' or 'K2' (default:'MI')
+        :return:
```
[the remark this answers is not in the document]
Comment (danwhale) [reply stable_gnn/model_gc.py]: Проверь  пожалуйста еще остальные функции, в некоторых местах осталась пустота

File: stable_gnn/embedding/sampling.py [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+    import torch_cluster  # noqa
+
+    RW = torch.ops.torch_cluster.random_walk
+except ImportError:
+    RW = None
+
+from torch_geometric.utils import subgraph
+
+
+class Sampler:
```
Comment (danwhale) [inline stable_gnn/embedding/sampling.py:29]: сделать класс абстрактным) вообще респект за такое!

File: stable_gnn/embedding/sampling.py [hunk 8]
```diff
@@ -0,0 +1,465 @@
...
+    """
+    Base class for sampling of positive and negative edges for unsupervised loss function
+
+    :param dataset_name: (str): Name of the inout dataset
+    :param data: (Graph): Input dataset
+    :param device: (device): Either 'cuda' or 'cpu'
+    :param loss_info: (dict): Dict of parameters of unsupervised loss function
+    """
+
+    def __init__(self, dataset_name: str, data: Graph, device: device, loss_info: Dict, **kwargs) -> None:
```
Comment (danwhale) [inline stable_gnn/embedding/sampling.py:39]: кейворд аргументы в либу не стоит вносить, нужно 100 проц понимать что происходит

Comment (danwhale) [reply stable_gnn/embedding/sampling.py:39]: лучше перегрузить классы наследники нужными аргументами)

File: stable_gnn/embedding/sampling.py [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+
+        x_new = x_new.tolist()
+
+        A = torch.zeros((len(x_new), len(x_new)), dtype=torch.long)
+        edge_index_0 = self.data.edge_index[0].tolist()
+        edge_index_1 = self.data.edge_index[1].tolist()
+        for j, i in enumerate(edge_index_0):
+            if i in x_new:
+                if edge_index_1[j] in x_new:
+                    A[i][edge_index_1[j]] = 1
```
Comment (danwhale) [inline stable_gnn/embedding/sampling.py:60]: Немного осталось, нужно добить все заглавные)

File: stable_gnn/embedding/sampling.py [hunk 10]
```diff
@@ -0,0 +1,465 @@
...
+    Sample positive and negative edges for APP unsupervised loss function
+
+    :param dataset_name: (str): Name of the inout dataset
+    :param data: (Graph): Input dataset
+    :param device: (device): Either 'cuda' or 'cpu'
+    :param mask: (Tensor): Tensor of True and False, True is on indices of nodes which are in the train set
+    :param loss_info: (dict): Dict of parameters of unsupervised loss function
+    """
+
+    def __init__(self, dataset_name: str, data: Graph, device: device, loss_info: Dict,mask = Tensor) -> None:
```
Comment (danwhale) [inline stable_gnn/embedding/sampling.py:398]: mask = Tensor вероятно тут нужно двоеточие

**Comment by danwhale** [comment]
Осталось немного, я так понимаю класс `TrainModel` мы пока намеренно не трогаем? 
Что из важного осталось сделать: 
- добавить ассерты в дженерал тест 
- добить все замечания `pflake8 .`, ну  TrainModel можно не править

File: stable_gnn/embedding/sampling.py [hunk 11]
```diff
@@ -36,45 +36,35 @@ class Sampler:
...
     """
 
-    def __init__(self, dataset_name: str, data: Graph, device: device, loss_info: Dict, **kwargs) -> None:
-        self.device = device
-        self.dataset_name = dataset_name
-        self.data = data.to(self.device)
-
-        self.negative_sampler = NegativeSampler(self.data, self.device)
-        self.loss = loss_info
-        super(Sampler, self).__init__()
```
Comment (danwhale) [inline stable_gnn/embedding/sampling.py]: Так а почему из абстрактного класса ты вынесла иницилизацию? тут почти все было ок)

File: tests/test_general_gc.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    model_training = TrainModelGC(
+        data=data,
+        conv=conv,
+        dataset_name=name,
+        device=device,
+        ssl_flag=ssl_flag,
+        extrapolate_flag=extrapolate_flag,
+    )
+
+    model, test_acc_mi, test_acc_ma = model_training.run(best_values)
```
Comment (danwhale) [inline tests/test_general_gc.py:42]: А тут нельзя ассерт добавить?

File: stable_gnn/embedding/sampling.py (hunk 11 above)
Comment (danwhale) [reply stable_gnn/embedding/sampling.py]: и логично, что классы-наследники просто сюда передают входные параметры, ну а только те, что нужны именно им иницилизируют дополнительно после иницилизации базового класса

File: tests/test_general_gc.py (hunk 12 above)
Comment (anpolol) [reply tests/test_general_gc.py:44]: я пока test_general_gc не проверяла, сейчас им займусь. Просто хотела побыстрее выложить что есть

Comment (danwhale) [reply tests/test_general_gc.py:44]: оки) хорошо) прост важно чтобы хотябы точность не просела, когда я начну немного ковырять код =)

File: stable_gnn/embedding/model_train_embeddings.py [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+                adjs = [adj.to(self.device) for adj in adjs]
+                out = model.forward(data.x[n_id.to(self.device)].to(self.device), adjs)
+                self._sampling(sampler, epoch, n_id[:batch_size], loss)
+                loss = model.loss(out, self.samples)
+                total_loss += loss
+        total_loss.backward()
+        optimizer.step()
+        return total_loss / len(train_loader), out
+
+    def run(self, params: Dict) -> Tensor:
```
Comment (danwhale) [inline stable_gnn/embedding/model_train_embeddings.py:68]: здесь params: Dict, в наследуемом просто число попыток передаем

File: stable_gnn/embedding/model_train_embeddings.py [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+            dropout=dropout,
+        )
+        model.to(self.device)
+
+        optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate, weight_decay=1e-5)
+
+        for epoch in range(99):
+            print(epoch)
+            loss, _ = self._train(
+                model,
```
Comment (danwhale) [inline stable_gnn/embedding/model_train_embeddings.py:101]: такого аргумента больше нет

File: stable_gnn/embedding/model_train_embeddings.py [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+                self.data,
+                optimizer,
+                loss_sampler,
+                train_loader,
+                dropout,
+                epoch,
```
Comment (danwhale) [inline stable_gnn/embedding/model_train_embeddings.py:107]: как и такого

**Review by danwhale — APPROVED** [review]
